tests/layers/vk_layer_logging.h

```diff
@@ -392,6 +392,10 @@ static inline bool debug_messenger_log_msg(const debug_report_data *debug_data,
 
     DebugAnnotFlagsToReportFlags(message_severity, message_type, &object_flags);
 
+    std::vector<VkDebugUtilsObjectNameInfoEXT> objects = std::vector<VkDebugUtilsObjectNameInfoEXT>(callback_data->objectCount);
```

Comment:
The approach of copying and repointing is going to be a maintenance/clarity challenge.  Better to construct the data in a non-const vector and point to it's data from the callback_data's const pointer